app/services/feeds/optimized.rb

```diff
@@ -0,0 +1,279 @@
+module Feeds
+  class Optimized
+    RANDOM_OFFSET_VALUES = [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13].freeze
```

Comment:
It's not clear to me why we have multiple instances of the same number in this constant, do you happen to know why?

Reply:
I asked this before: we generate a random index into this array, so this skews the probabilities in favor of certain numbers.